client/my-sites/email/email-providers-in-depth-comparison/comparison-table/index.tsx

```diff
@@ -1,22 +1,59 @@
 /* eslint-disable wpcalypso/jsx-classname-namespace */
 
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useSelector } from 'react-redux';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 import type {
 	ComparisonTableProps,
 	EmailProviderFeatures,
 } from 'calypso/my-sites/email/email-providers-in-depth-comparison/types';
 import type { ReactElement } from 'react';
 
-const ComparisonTable = ( { emailProviders }: ComparisonTableProps ): ReactElement => {
+const ComparisonTable = ( {
+	emailProviders,
+	intervalLength,
+	selectedDomainName,
+}: ComparisonTableProps ): ReactElement => {
+	const translate = useTranslate();
+
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const currentRoute = useSelector( getCurrentRoute );
+
+	const selectEmailProvider = ( emailProviderSlug: string ) => {
+		if ( selectedSiteSlug === null ) {
+			return;
+		}
+
+		page(
+			emailManagementPurchaseNewEmailAccount(
+				selectedSiteSlug,
+				selectedDomainName,
+				currentRoute,
+				null,
+				emailProviderSlug,
+				intervalLength
+			)
+		);
+	};
+
 	return (
 		<div className="email-providers-in-depth-comparison-table">
-			{ emailProviders.map( ( emailProviderFeatures: EmailProviderFeatures, index: number ) => {
+			{ emailProviders.map( ( emailProviderFeatures: EmailProviderFeatures ) => {
 				return (
-					<div key={ index }>
+					<div key={ emailProviderFeatures.slug }>
```

Comment:
It may be worth including `intervalLength` in the key to make sure we don't trigger collisions when toggling between monthly and annual. I think we _might_ be OK without that, but I am not confident!
```suggestion
					<div key={ emailProviderFeatures.slug + '-' + intervalLength }>
```

Reply:
I'll keep that in mind when I add monthly/annual prices in the next pull request.